cmake/CMakeLists.txt

```diff
@@ -473,6 +474,19 @@ if (onnxruntime_USE_MKLDNN)
   link_directories(${MKLDNN_LIB_DIR})
 endif()
 
+if (onnxruntime_USE_NGRAPH)
+  if (Win32)
+    message(FATAL_ERROR "nGraph is not currently supported on Windows.")
+  endif()
+  if (onnxruntime_USE_OPENMP)
+    message(FATAL_ERROR "Please set onnxruntime_USE_OPENMP=OFF for nGraph execution provider.")
+  endif()
+  add_definitions(-DUSE_NGRAPH=1)
```

Comment:
should we also check that we are using FULL protobuf option here?

Reply:
Yes, it needs to be here. I'll add it.

Reply:
Also, kind of related question: How do you suggest, I get around this hardcoded ```use_openmp``` here. https://github.com/Microsoft/onnxruntime/blob/3a8b9a4/build.sh#L9

Reply:
Added check for FULL protobuf.